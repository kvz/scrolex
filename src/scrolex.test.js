const utils    = require('./utils')
const test     = require('ava')
// const debug = require('depurar')('sut')
const sut      = utils

const cacheDir      = 'CACHEDIR'
const projectDir    = 'PROJECTDIR'
const lanyonVersion = 'LANYONVERSION'

test('dockerCmd', (t) => {
  const res = sut.dockerCmd({cacheDir, projectDir, lanyonVersion}, 'CMD', 'FLAGS')
  t.is(res, 'docker run FLAGS --rm --workdir CACHEDIR --user $(id -u) --volume CACHEDIR:CACHEDIR --volume PROJECTDIR:PROJECTDIR kevinvz/lanyon:LANYONVERSION CMD', 'should return valid dockerCmd')
})

// test.cb('inbox', (t) => {
//   const event   = 'ambient'
//   const bot     = sut._bot
//   const message = { text: 'Hey Botty' }
//   sut.inbox(event, bot, message, (err, response) => {
//     t.ifError(err, 'should respond without error')
//     t.regex(response.text, /Hi mock-friend!/, 'response should match "Hi mock-friend!"')
//     t.end()
//   })
// })
