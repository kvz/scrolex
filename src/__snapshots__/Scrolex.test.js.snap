exports[`Scrolex _defaults should default according to the snapshot 1`] = `
Object {
  "addCommandAsComponent": false,
  "announce": false,
  "cbPreLinefeed": [Function],
  "cleanupTmpFiles": true,
  "components": "a>b>c",
  "cwd": "/tmp",
  "dryrun": false,
  "fatal": false,
  "indent": 4,
  "interval": 80,
  "mode": "singlescroll",
  "shell": false,
  "tmpFiles": false,
}
`;

exports[`Scrolex _normalizeOpts should normalize according to the snapshot 1`] = `
Object {
  "components": Array [
    "a",
    "b",
    "c",
  ],
  "indent": 0,
  "mode": "singlescroll",
}
`;

exports[`ScrolexExports exe should accept async/await 1`] = `
"

    Starting with limit=1, interval=1000, errorOut=true
    Doing thing 1. fakecmd with limit=1, interval=1000, errorOut=true done.

. Fault while executing \"MOCK_ERROR_OUT=1 MOCK_LIMIT=1 node #{SCROLEX_ROOT}/src/fakecmd.js\". Exit code: 1"
`;

exports[`ScrolexExports exe should accept async/await 2`] = `
"Starting with limit=2, interval=1000, errorOut=false
Doing thing 1
Doing thing 2. fakecmd with limit=2, interval=1000, errorOut=false done."
`;

exports[`ScrolexExports exe should accept async/await 3`] = `
"Starting with limit=2, interval=1000, errorOut=false
Doing thing 1
Doing thing 2. fakecmd with limit=2, interval=1000, errorOut=false done."
`;

exports[`ScrolexExports exe should accept async/await 4`] = `
"Starting with limit=2, interval=1000, errorOut=false
Doing thing 1
Doing thing 2. fakecmd with limit=2, interval=1000, errorOut=false done."
`;

exports[`ScrolexExports exe should accept async/await and catch an error 1`] = `
[Error: 

    Starting with limit=1, interval=1000, errorOut=true
    Doing thing 1. fakecmd with limit=1, interval=1000, errorOut=true done.

. Fault while executing "MOCK_ERROR_OUT=1 MOCK_LIMIT=1 node #{SCROLEX_ROOT}/src/fakecmd.js". Exit code: 1]
`;
